### PR TITLE
Catch error in supported isolators

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Add a `try...except` around the check for supported isolators

## Why are we making these changes?
Improves test reliability

### CODE REVIEW

File: integration/tests/cook/util.py [hunk 1]
```diff
@@ -1312,11 +1312,15 @@ def _supported_isolators():
     mesos_url = retrieve_mesos_url()
     mesos_state = get_mesos_state(mesos_url)
     slave_endpoint = get_agent_endpoint(mesos_state, mesos_state['slaves'][0]['hostname'])
-    slave_state = session.get(slave_endpoint).json()
-    if 'flags' in slave_state:
-        return set(slave_state['flags']['isolation'].split(','))
-    else:
-        logger.error(f'Unable to determine flags from slave state at {slave_endpoint}: {slave_state}')
+    try:
```
Comment (scrosby) [inline integration/tests/cook/util.py]: Can you put a 'why' here:
""The request can be empty, bad json, etc.""

Two related questions:
  If its OK to return [] on an error, why not return [] all the time?
  If errors aren't good, should we be trying more than once here?

File: integration/tests/cook/util.py [hunk 2]
```diff
@@ -1312,11 +1312,15 @@ def _supported_isolators():
...
+        slave_state = session.get(slave_endpoint).json()
+        if 'flags' in slave_state:
+            return set(slave_state['flags']['isolation'].split(','))
+        else:
+            logger.error(f'Unable to determine flags from slave state at {slave_endpoint}: {slave_state}')
+            return []
+    except Error as e:
+        logger.exception('Unable to parse json response')
```
Comment (scrosby) [inline integration/tests/cook/util.py]: Probably worth printing what the response was here and the error.

Comment (pschorf) [reply integration/tests/cook/util.py]: Done

**Comment by scrosby** [comment]
Looks good to me. Just want to see a green travis build.

**Comment by pschorf** [comment]
Green!

**Pull request merged**